src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -1152,6 +1155,21 @@ internal void InitializeNamespacesToExcludeFromLibrary()
             }
         }
 
+        /// <summary>
+        /// Update namespaces to exclude from Library based on feature flags
+        /// </summary>
+        internal void UpdateNamespacesToExcludeFromLibrary()
+        {
+            // When the experiment toggle is disabled by feature flag, include the TSpline namespace from the library OOTB.
+            if (!DynamoModel.FeatureFlags?.CheckFeatureFlag("IsTSplineNodesExperimentToggleVisible", false) ?? false)
```

Comment:
Why not use `preferenceSettings.IsTSplineNodesExperimentToggleVisible` property here?